assignments/homework/Raymond Ingram/Homework 2.py

```diff
@@ -0,0 +1,313 @@
+
+# In[7]:
+
+
+hdu2 = Table.read('data/gaia_lmc_psc.fits')
+hdu2
+
+
+# In[8]:
+
+
+_ = plt.hist(hdu2['phot_g_mean_mag'], bins=100)
+
+
+# In[9]:
+
+
+coord1 = SkyCoord(hdu2['ra']*u.deg, hdu2['dec']*u.deg)
+coord2 = coord1.transform_to('galactic')
+coord2
+plt.scatter(coord2.l,coord2.b)
+
+
+# In[10]:
+
+
+hdu = fits.PrimaryHDU()
+hdu
+
+
+# In[11]:
+
+
+hdu.table = hdu2
+dataz = fits.getdata('data/gaia_lmc_psc.fits')
+hdu2.data = dataz
+
+
+# In[12]:
+
+
+hdu.writeto('newFits1.fits', overwrite=True)
+
+
+# # <section class="challenge panel panel-success">
+# <div class="panel-heading">
+# <h2><span class="fa fa-pencil"></span> Challenge</h2>
+# </div>
+# 
+# 
+# <div class="panel-body">
+# 
+# <ol>
+# <li>Make a table that contains three columns: <code>spectral type</code>, <code>temperature</code>, and <code>radius</code>, and incude 5 rows with fake data (or real data if you like, for example from <a href="http://www.atlasoftheuniverse.com/startype.html">here</a>). Try including units on the columns that can have them.</li>
+# <li>Find the mean temperature and the maximum radius</li>
+# <li>Try and find out how to add and remove rows</li>
+# <li>Add a new column which gives the luminosity (using $L=4\pi R^2 \sigma T^4$)</li>
+# </ol>
+# 
+# </div>
+# 
+# </section>
+# 
+
+# In[13]:
+
+
+import astropy.constants as const
+
+
+# In[14]:
+
+
+table1 = Table()
+table1['Spectral Type'] = ['A','B','C','D','E']
+table1['Temperature'] = [100,2213,321,4131,5241]*u.K
+table1['Radius'] = [5, 21, 2, 34, 12]*const.R_sun
+table1
+
+
+# In[15]:
+
+
+np.mean(table1['Temperature']), np.max(table1['Radius'])
+
+
+# In[16]:
+
+
+table1.add_row(['F', 798*u.K, 6])
+table1
+
+
+# In[17]:
+
+
+table1['Luminosity']=4*np.pi*const.sigma_sb*[table1['Radius']**2+table1['Temperature']**4]
+table1
+
+
+# # <section class="challenge panel panel-success">
+# <div class="panel-heading">
+# <h2><span class="fa fa-pencil"></span> Challenge</h2>
+# </div>
+# 
+# 
+# <div class="panel-body">
+# 
+# <p>Starting from the <code>obs</code> table:</p>
+# <ol>
+# <li>Make a new table that shows every other row, starting with the second row? (that is, the second, fourth, sixth, etc. rows).</li>
+# <li>Make a new table the only contains rows where <code>name</code> is <code>M31</code></li>
+# </ol>
+# 
+# </div>
+# 
+# </section>
+# 
+
+# In[18]:
+
+
+obs = Table(rows=[('M31' , '2012-01-02', 17.0, 17.5),
+                  ('M31' , '2012-01-02', 17.1, 17.4),
+                  ('M101', '2012-01-02', 15.1, 13.5),
+                  ('M82' , '2012-02-14', 16.2, 14.5),
+                  ('M31' , '2012-02-14', 16.9, 17.3),
+                  ('M82' , '2012-02-14', 15.2, 15.5),
+                  ('M101', '2012-02-14', 15.0, 13.6),
+                  ('M82' , '2012-03-26', 15.7, 16.5),
+                  ('M101', '2012-03-26', 15.1, 13.5),
+                  ('M101', '2012-03-26', 14.8, 14.3)],
+            names=['name', 'obs_date', 'mag_b', 'mag_v'])
+
+
+# In[19]:
+
+
+obs2 = obs[2::2]
```

Comment:
Python starts indexing from 0, so here you did every other row starting from the third one